src/components/EventForm.js

```diff
@@ -78,21 +79,18 @@ export default class EventForm extends Component {
   onSubmit = async (values, { resetForm }) => {
     let response;
     try {
-      const user = JSON.parse(await asyncStore.retrieve("user"));
+      const user = this.props.user;
       if (!user) throw new Error("Not logged in.");
-      // Add username so created.by.user user id gets added on server:
-      let data = {
-        ...values,
-        user: { username: user.username }
-      }
-      // Don't pass in data.updated so new date gets added on server:
-      if (data.updated) delete data["updated"];
+
+      let data = { ...values }
 
       if (this.props.newEvent === true) {
         response = await eventServices.post(data);
       } else {
         // If updating event, add event _id, fix/remove server-generated data
         if (data.created) delete data["created"];
+        if (data.updated) delete data["updated"];
+        // If updating event, add event _id
```

Comment:
I don't think we need this extra comment here.

Reply:
Fixing this as well-- the id key's underscore is no longer removed on the server side anyway.